scripts/check-diff

```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+
+if [[ "$(git diff --exit-code)" || ! -z "$(git ls-files -o --exclude-standard)" ]]
+then
+    echo "Mismatches found between .riot/requirements/*.txt and riotfile.py. Run scripts/compile-and-prune-test-requirements and commit the result."
```

Comment:
what are your thoughts on making this script more generic, e.g. even just:

```shell
#!/bin/bash

[[ "$(git diff --exit-code)" || ! -z "$(git ls-files -o --exclude-standard)" ]]
```

Since the error message is very specific to that one workflow, but the script name/check is fairly generic.



or could do:

```shell
#!/bin/bash

message=${1:"Unexpected files changed or added"}

if [[ "$(git diff --exit-code)" || ! -z "$(git ls-files -o --exclude-standard)" ]]
then
    echo "${message}"
    exit 1
fi
```


I know I am probably being really pedantic here, so feel free to take this as an optional suggestion.

Reply:
I think that's a good idea.